src/components/data/DataTable.js

```diff
@@ -410,18 +410,13 @@ const DataTable = props => {
       },
       onDismiss: () => setUpdatingEntity(undefined)
     }),
-    !!deletingColumn && h(PromptedConfirmationModal, {
-      title: 'Delete Column',
-      confirmationPrompt: 'Delete column',
-      buttonText: 'Delete column',
+    !!deletingColumn && h(DeleteConfirmationModal, {
+      objectType: 'column',
+      objectName: deletingColumn,
       onConfirm: () => deleteColumn(deletingColumn),
       onDismiss: () => setDeletingColumn(undefined)
-    }, [
-      div(['Are you sure you want to permanently delete the column ',
-        span({ style: { fontWeight: 600, wordBreak: 'break-word' } }, deletingColumn), '?']),
-      div({ style: { fontWeight: 500, marginTop: '1rem' } }, 'This cannot be undone.')
-    ]),
-    !!clearingColumn && h(PromptedConfirmationModal, {
+    }),
+    !!clearingColumn && h(DeleteConfirmationModal, {
```

Comment:
What was the goal of moving from `PromptedConfirmationModal` to the delete one? It's pretty confusing to call something called "DeleteConfirmationModal" to do an operation other than deleting.

It seems like you have pretty much made `PromptedConfirmationModal` unnecessary, and the 2 things should just be collapsed into 1.

Reply:
The goal of [SU-56](https://broadworkbench.atlassian.net/browse/SU-56) is a consistent style of confirmation prompt for deleting things. Though the label here is "Clear column", it is deleting data from the table.

The reason for a component specific to confirming deletes is that, once more confirmation prompts are moved to `DeleteConfirmationModal`, `DeleteConfirmationModal` will get a yellow/warning color background, which wouldn't really make sense for a generic confirmation prompt.

Now, `PromptedConfirmationModal` exists as a slightly more generic version of `DeleteConfirmationModal`, but since it is not used anywhere else, it could be absorbed into `DeleteConfirmationModal`.